src/modules/Chat/components/ChatList/index.ts

```diff
@@ -0,0 +1,135 @@
+        phone: string,
+    }
+    addChatIconSrc: string,
+    searchIconSrc: string,
+    styles?: string
+}
+
+export class ChatList extends Block {
+  constructor(props: ChatListProps) {
+    super('div', props);
+        this.element!.classList.add(chatListStyles.wrapper);
+  }
+
+  protected editPropsBeforeMakeThemProxy(props: ChatListProps) {
+    props.styles = chatListStyles;
+  }
+
+  protected init() {
+    this._addAvatar();
+    this._addButton();
+    this._addSearchInput();
+    this._addItemsList();
+    this._addChatUser();
+  }
+
+  protected render() {
+    return this.compile(template, this.props);
+  }
+
+  private _addChatUser() {
+    this.children.chatUser = new ChatUser({
+      avatar: new Avatar({
+        src: this.props.profile.avatarSrc,
+        size: '2em',
+        alt: this.props.profile.login,
+        title: this.props.profile.login,
+      }),
+      login: this.props.profile.login,
+      events: {
+        click: () => {
+          PageNavigator.renderProfilePage();
+        },
+      },
+    });
+  }
+
+  private _addAvatar() {
+    this.children.avatar = new Avatar({
+      src: this.props.profile.avatarSrc,
+      size: '2em',
+      alt: this.props.profile.login,
+      title: this.props.profile.login,
+    });
+  }
+
+  private _addButton() {
+    this.children.addButton = new Icon({
+      size: '1.5em',
+      icon: this.props.addChatIconSrc,
+    });
+        this.children.addButton.element!.addEventListener('click', () => {
+          console.log('Нажата кнопка добавления нового чата');
+        });
+  }
+
+  private _addSearchInput() {
+    this.children.searchInput = new Input({
+      title: 'Поиск',
+      type: 'text',
+      name: 'search',
+      value: '',
+      placeholder: 'Поиск',
+      isRounded: true,
+      isLight: true,
+      displayBlock: true,
+      iconSrc: this.props.searchIconSrc,
+    });
+  }
+
+  private _addItemsList() {
+    const itemsList: ChatItem[] = [];
+    this._testItemsData().forEach((element: ChatItemProps) => {
+      itemsList.push(new ChatItem(element));
+    });
+    this.children.chatItemsList = itemsList;
+  }
+
+  private _testItemsData():ChatItemProps[] {
+    const itemsData: ChatItemProps[] = [];
+    let notRead = 0;
+    for (let i = 1; i <= 20; i++) {
+      const min = Math.ceil(-80);
+      const max = Math.floor(80);
+      notRead = Math.floor(Math.random() * (max - min + 1)) + min;
+
+      let imageSrc = '';
+      if (i < 10) {
+        imageSrc = `/upload/img/cat_0${i}.jpg`;
+      } else {
+        imageSrc = `/upload/img/cat_${i}.jpg`;
+      }
+      itemsData.push({
+        chatId: `${i}`,
+        imageSrc,
+        name: `Чат с номером ${i}`,
+        message: {
+          text: 'Случайный текст для проверки отображение его в списке чатов. '
+                        + 'Случайный текст для проверки отображение его в списке чатов.',
+          dateTime: '20:43',
+        },
+        notRead: notRead > 0 ? notRead : 0,
```

Comment:
а что, если сделать notRead типа boolean?
В таком случае выражение упростится до такого:
```suggestion
        notRead
```

Reply:
Оно хранит в себе именно количество непрочитанных сообщений, а не сам факт непочтения, и чтобы не создавать два поля одно булево другое с количеством, было принято решение написать такую проверку